stdlib/fun.mli

```diff
@@ -61,3 +61,49 @@ exception Finally_raised of exn
     an unexpected exception or a programming error. As a general rule,
     one should not catch a [Finally_raised] exception except as part of
     a catch-all handler. *)
+
+val with_resource :
+  acquire:('a -> 'r) -> 'a -> scope:('r -> 'b) -> release:('r -> unit) -> 'b
+(** [with_resource ~acquire x work ~release] invokes [acquire x], then
```

Comment:
Should be: `[with_resource ~acquire x ~scope ~release]` to match the type signature.